fastestimator/trace/metric/precision.py

```diff
@@ -27,37 +28,16 @@ class Precision(Trace):
     Args:
         true_key: Name of the keys in the ground truth label in data pipeline.
         pred_key: Name of the keys in predicted label. Defaults to None.
-        labels: The set of labels to include. For more details please refer tosklearn.metrics.precision_score. Defaults
-            to None.
-        pos_label: The class to report. For more details please refer to sklearn.metrics.precision_score. Defaults to 1.
-        average: It should be one of {"auto", "binary", "micro", "macro", "weighted", "samples", None}. If "auto", the
-            trace will detect the input data type and choose the right average for you. Otherwise, it will pass its to
-            sklearn.metrics.precision_score. Defaults to "auto".
-        sample_weight: Sample weight. If None, it will not apply sample weight. For more details please refer to
-            sklearn.metrics.precision_score. Defaults to None.
         mode: Restrict the trace to run only on given modes {'train', 'eval', 'test'}. None will always execute.
             Defaults to 'eval'.
         output_name: Name of the key to store to the state. Defaults to "precision".
     """
-
-    y_true: List[int]
-    y_pred: List[int]
-
     def __init__(self,
                  true_key: str,
                  pred_key: str,
-                 labels: Optional[list] = None,
-                 pos_label: Union[int, str] = 1,
-                 average: str = 'auto',
```

Comment:
why are we dropping all this functionality?

Reply:
some doesn't apply to us, some are unnecessary.

Reply:
why would things like sample weighting be unnecessary? Those params exist because people use them...

Reply:
users are expected to provide weight of all samples together. in deep learning, it's not practical to provide that list (consider having to create a list with 1.3million element for ImageNet).

If we are going to do it in the future, we may have to read from batch dictionary instead of providing that array upfront.

Reply:
fair enough. don't we still need pos_label though? since precision is not an asymmetric metric?